tests/scaletest/scale_test.go

```diff
@@ -391,78 +574,285 @@ func DeleteIngressesSerial(t *testing.T, numOfIng int, initialNumOfPools int, Av
+		ingressesCreated, _, err = lib.CreateInsecureIngress(ingressNamePrefix, listOfServicesCreated[0], namespace, deletionStartPoint)
+		if err != nil {
+			t.Fatalf("Failed to create %s ingresses as : %v", ingressType, err)
+		}
+	case ingressType == SECURE:
+		t.Logf("Creating %d %s Ingresses...", deletionStartPoint, ingressType)
+		ingressesCreated, _, err = lib.CreateSecureIngress(ingressNamePrefix, listOfServicesCreated[0], namespace, deletionStartPoint)
+		if err != nil {
+			t.Fatalf("Failed to create %s ingresses as : %v", ingressType, err)
+		}
+	case ingressType == MULTIHOST:
+		t.Logf("Creating %d %s Ingresses...", deletionStartPoint, ingressType)
+		ingressesCreated, _, _, err = lib.CreateMultiHostIngress(ingressNamePrefix, listOfServicesCreated, namespace, deletionStartPoint)
+		if err != nil {
+			t.Fatalf("Failed to create %s ingresses as : %v", ingressType, err)
+		}
+	}
+	wg.Add(3)
+	go HybridCreation(t, &wg, numOfIng, deletionStartPoint)
+	go HybridUpdation(t, &wg, numOfIng/2)
+	go HybridDeletion(t, &wg, numOfIng)
+	wg.Wait()
+	g.Expect(ingressesCreated).To(gomega.HaveLen(numOfIng))
+	g.Expect(ingressesDeleted).To(gomega.HaveLen(numOfIng))
+}
+
+func CreateIngressParallelWithAkoReboot(t *testing.T) {
+	REBOOTAKO = true
+	CreateIngressesParallel(t, numOfIng, initialNumOfPools)
+	REBOOTAKO = false
+}
+
+func CreateIngressParallelWithControllerReboot(t *testing.T) {
+	REBOOTCONTROLLER = true
+	CreateIngressesParallel(t, numOfIng, initialNumOfPools)
+	REBOOTCONTROLLER = false
+}
+
+func CreateIngressParallelWithNodeReboot(t *testing.T) {
+	REBOOTNODE = true
+	CreateIngressesParallel(t, numOfIng, initialNumOfPools)
+	REBOOTNODE = false
+}
+
+func DeleteIngressParallelWithAkoReboot(t *testing.T) {
+	REBOOTAKO = true
+	DeleteIngressesParallel(t, numOfIng, initialNumOfPools)
+	REBOOTAKO = false
+}
+
+func DeleteIngressParallelWithControllerReboot(t *testing.T) {
+	REBOOTCONTROLLER = true
+	DeleteIngressesParallel(t, numOfIng, initialNumOfPools)
+	REBOOTCONTROLLER = false
+}
+
+func DeleteIngressParallelWithNodeReboot(t *testing.T) {
+	REBOOTNODE = true
+	DeleteIngressesParallel(t, numOfIng, initialNumOfPools)
+	REBOOTNODE = false
+}
+
+func UpdateIngressParallelWithAkoReboot(t *testing.T) {
+	REBOOTAKO = true
+	UpdateIngressesParallel(t, numOfIng)
+	REBOOTAKO = false
+}
+
+func UpdateIngressParallelWithControllerReboot(t *testing.T) {
+	REBOOTCONTROLLER = true
+	UpdateIngressesParallel(t, numOfIng)
+	REBOOTCONTROLLER = false
+}
+
+func UpdateIngressParallelWithNodeReboot(t *testing.T) {
+	REBOOTNODE = true
+	UpdateIngressesParallel(t, numOfIng)
+	REBOOTNODE = false
 }
 
 func TestMain(t *testing.M) {
 	Setup()
-	t.Run()
-	Cleanup()
+	// t.Run()
+	// Cleanup()
 }
 
-func TestParallel200InsecureIngresses(t *testing.T) {
+func TestInsecureParallelCreationUpdationDeletionWithoutReboot(t *testing.T) {
+	SetupForTesting(t)
 	ingressType = INSECURE
-	ParallelIngressHelper(t, 200)
+	CreateIngressesParallel(t, numOfIng, initialNumOfPools)
```

Comment:
The variable ingressType is used by 14 functions in total. So I think it would be better to keep it as a global variable. Should I still send it as an argument to functions?